modules/pinger/src/main/java/org/hawkular/component/pinger/Traits.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.component.pinger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.apache.http.Header;
+import org.apache.http.HeaderIterator;
+import org.apache.http.HttpResponse;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * A collection of traits retrieved from response headers. The headers that are considered to be interesting are listed
+ * in {@link TraitHeader} enum. Trait is an interesting piece of information about a monitored site, such as the name of
+ * the web server implementation that is serving the site, the technology used in the serving application, etc.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Traits {
+
+    /**
+     * The enumeration of HTTP response headers that are considered to bear interesting information about a monitored
+     * site.
+     */
+    public enum TraitHeader {
+        SERVER("server"), X_ASPNET_VERSION("x-aspnet-version"), X_POWERED_BY("x-powered-by"), X_RUNTIME("x-runtime"),
+        X_VERSION("x-version");
```

Comment:
We can add more later, what is interesting in our context for now, is to know:
X-Powered-By: Undertow/1
Server: WildFly/8